### ISSUE TITLE
Softmax Implementation Broken

### ISSUE THREAD

**Issue by jdk514** [jdk514/gwu_nn#3]
Currently softmax backprop is broken. Implementation is not correctly handling how the softmax partial derivative functions and needs to be updated.
